openapi_python_client/parser/openapi.py

```diff
@@ -232,6 +233,16 @@ def _add_parameters(
             )
             if isinstance(prop, ParseError):
                 return ParseError(detail=f"cannot parse parameter of endpoint {endpoint.name}", data=prop.data), schemas
+
+            if prop.python_name in used_python_names:
+                prop.set_python_name(f"{prop.python_name}_{param.param_in}")
+
+            if prop.python_name in used_python_names:
+                return (
+                    ParseError(detail=f"Encountered duplicate properties named {prop.python_name}", data=data),
+                    schemas,
+                )
+            used_python_names.add(prop.python_name)
```

Comment:
```suggestion

            if prop.python_name in used_python_names:
                duplicate_prop, duplicate_prop_param_location = used_python_names[prop.python_name]
                if duplicate_prop.python_name == prop.python_name:
                    duplicate_prop.set_python_name(f"{duplicate_prop.python_name}_{duplicate_prop_param_location}")
                
                prop.set_python_name(f"{prop.python_name}_{param.param_in}")

            if prop.python_name in used_python_names:
                return (
                    ParseError(detail=f"Encountered duplicate properties named {prop.python_name}", data=data),
                    schemas,
                )
            used_python_names[prop.python_name] = (prop, param.param_in)
```

Reply:
Good call, that should make it much more deterministic too, so switching the param order in the OpenAPI document doesn't break API clients.